### PR TITLE
Do change speed support

### PR DESCRIPTION
**Pull request by LorenzMeier**

### CODE REVIEW

**Comment by LorenzMeier** [comment]
@MikaelFerland  @tubeme This is now implemented, but UNTESTED. Simulation testing would be appreciated. I will get to test it later this weekend.

**Comment by LorenzMeier** [comment]
Support is in for QGC. Just needs to be validated here.

**Comment by LorenzMeier** [comment]
Doesn't seem to work yet. Not sure why.

**Comment by MikaelFerland** [comment]
Does it take the speed from an average of GPS data or it uses the ground
speed sensor as reference? I have tested over HITL with jMAVSim, but I was
unsuccesful for me also.

Doesn't seem to work yet. Not sure why.

—
Reply to this email directly or view it on GitHub
<https://github.com/PX4/Firmware/pull/3567#issuecomment-182807771>.

**Comment by tubeme** [comment]
Does not work in mission. Could not load the mission QGC gave error. I'm using the Latest master with the Intel angle patch from couple of days. PX4 master is 5 days old. 

setspeed

**Comment by LorenzMeier** [comment]
@tubeme I've tested it after a few additional fixes successfully in SITL. @dagar Could you test with Gazebo / VTOL sim in fixed wing mode? Then it would be ready for flight tests.

screen shot 2016-02-17 at 16 49 19

**Comment by MikaelFerland** [comment]
@LorenzMeier Did you see my comments? Addionally Do I will be able to test it with jMAVSim or you guys usually tested it with Gazebo. Because it will fix my setup to get better result. Thank you for your time.

**Comment by LorenzMeier** [comment]
@MikaelFerland You would need to re-test with the current state I pushed. Its ground speed based for multicopters, currently airspeed based for planes. We'll clean this up for planes later.

**Comment by MikaelFerland** [comment]
Perfect, to simulate it jMAVSim or Gazebo is prefered?

**Comment by LorenzMeier** [comment]
Both works.

**Comment by dagar** [comment]
It seems my old thinkpad is nowhere near fast enough to run the gazebo simulation, so you may have finally given me a reason to upgrade. I'll try it again on a faster machine tomorrow.

**Comment by LorenzMeier** [comment]
Rebased on master. Needs flight testing. Tested successfully in SITL in multicopter mode.

**Comment by tubeme** [comment]
Do I need the latest master of QGC for this mission or can go with the current version I use from the screenshot?

**Comment by dagar** [comment]
screenshot from 2016-02-18 20 09 31

I'm getting airspeed errors in the gazebo simulation now.

**Comment by dagar** [comment]
Rebased on master to get airspeed fix.

File: src/modules/mc_pos_control/mc_pos_control_params.c [hunk 1]
```diff
@@ -236,16 +236,28 @@ PARAM_DEFINE_FLOAT(MPC_XY_VEL_I, 0.02f);
...
- * Maximum horizontal velocity in AUTO mode and endpoint for position stabilized mode (POSCTRL).
+ * Maximum horizontal velocity in AUTO mode.
  *
  * @unit m/s
  * @min 0.0
  * @decimal 2
  * @group Multicopter Position Control
  */
-PARAM_DEFINE_FLOAT(MPC_XY_VEL_MAX, 5.0f);
+PARAM_DEFINE_FLOAT(MPC_XY_VEL_MAX, 20.0f);
```
Comment (AndreasAntener) [inline src/modules/mc_pos_control/mc_pos_control_params.c]: Really? ;) I'd say 10 would be the max acceptable default.

File: src/modules/navigator/navigator_main.cpp [hunk 2]
```diff
@@ -151,7 +151,10 @@ Navigator::Navigator() :
 	_param_loiter_radius(this, "LOITER_RAD"),
 	_param_acceptance_radius(this, "ACC_RAD"),
 	_param_datalinkloss_obc(this, "DLL_OBC"),
-	_param_rcloss_obc(this, "RCL_OBC")
+	_param_rcloss_obc(this, "RCL_OBC"),
+	_param_cruising_speed_hover(this, "MPC_XY_VEL_MAX", false),
```
Comment (AndreasAntener) [inline src/modules/navigator/navigator_main.cpp]: I guess that should be MPC_XY_CRUISE

Comment (LorenzMeier) [inline src/modules/navigator/navigator_main.cpp]: Yes indeed.

**Comment by LorenzMeier** [comment]
Merged to master.

**Comment by MikaelFerland** [comment]
@LorenzMeier Thank you, we tested it last weekend, and it works very well. We land unsuccessfully, but I think we broke the power plug before the mission start. We will test it again this sunday of the weather permit us.

**Pull request merged**